src/chainstate/burn/operations/peg_in.rs

```diff
@@ -0,0 +1,451 @@
+
+        match op {
+            Err(OpError::ParseError) => (),
+            result => panic!("Expected OpError::ParseError, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_in_should_return_error_given_no_second_output() {
+        let mut rng = seeded_rng();
+        let opcode = Opcodes::PegIn;
+
+        let mut data = vec![1];
+        let addr_bytes: [u8; 20] = random_bytes(&mut rng);
+        data.extend_from_slice(&addr_bytes);
+
+        let tx = burnchain_transaction(data, None, opcode);
+        let header = burnchain_block_header();
+
+        let op = PegInOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::InvalidInput) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_in_should_return_error_given_too_short_data_array() {
+        let mut rng = seeded_rng();
+        let opcode = Opcodes::PegIn;
+
+        let peg_wallet_address = random_bytes(&mut rng);
+        let amount = 10;
+        let output2 = Output2Data::new_as_option(amount, peg_wallet_address);
+
+        let mut data = vec![1];
+        let addr_bytes: [u8; 19] = random_bytes(&mut rng);
+        data.extend_from_slice(&addr_bytes);
+
+        let tx = burnchain_transaction(data, output2, opcode);
+        let header = burnchain_block_header();
+
+        let op = PegInOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::ParseError) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_check_should_return_error_on_zero_amount_and_ok_on_any_other_values() {
+        let mut rng = seeded_rng();
+
+        let peg_wallet_address = random_bytes(&mut rng);
+
+        let mut data = vec![1];
+        let addr_bytes = random_bytes(&mut rng);
+        let stx_address = StacksAddress::new(1, addr_bytes.into());
+        data.extend_from_slice(&addr_bytes);
+
+        let create_op = move |amount| {
+            let opcode = Opcodes::PegIn;
+            let output2 = Output2Data::new_as_option(amount, peg_wallet_address.clone());
+
+            let tx = burnchain_transaction(data.clone(), output2, opcode);
+            let header = burnchain_block_header();
+
+            PegInOp::from_tx(&header, &tx).expect("Failed to construct peg-in operation")
+        };
+
+        match create_op(0).check() {
+            Err(OpError::PegInAmountMustBePositive) => (),
+            result => panic!(
+                "Expected OpError::PegInAmountMustBePositive, got {:?}",
+                result
+            ),
+        };
+
+        create_op(1)
+            .check()
+            .expect("Any strictly positive amounts should be ok");
+
+        create_op(u64::MAX)
+            .check()
+            .expect("Any strictly positive amounts should be ok");
+    }
+
+    // TODO: Invalid contract name test
```

Comment:
Please complete this task before merging.

Reply:
Done. I had already added the test case, but forgot to clean up the stale todo.